src/charm.py

```diff
@@ -1,104 +1,296 @@
+        triggers start of the alertmanager-configurer service.
 
-        Learn more about config at https://juju.is/docs/sdk/config
+        Args:
+            event: Juju PebbleReadyEvent event
         """
-        current = self.config["thing"]
-        if current not in self._stored.things:
-            logger.debug("found a new thing: %r", current)
-            self._stored.things.append(current)
+        if not self.model.get_relation("alertmanager"):
+            self.unit.status = BlockedStatus("Waiting for alertmanager relation to be created")
+            event.defer()
+            return
+        if not self._alertmanager_configurer_container.can_connect():
+            self.unit.status = WaitingStatus(
+                f"Waiting for {self._alertmanager_configurer_container_name} container to be ready"
+            )
+            event.defer()
+            return
+        if not self._dummy_http_server_running:
+            self.unit.status = WaitingStatus("Waiting for the dummy HTTP server to be ready")
+            event.defer()
+            return
+        self._start_alertmanager_configurer()
+        self.unit.status = ActiveStatus()
 
-    def _on_fortune_action(self, event):
-        """Just an example to show how to receive actions.
+    def _on_dummy_http_server_pebble_ready(self, event: PebbleReadyEvent) -> None:
+        """Event handler for dummy-http-server pebble ready event.
 
-        TEMPLATE-TODO: change this example to suit your needs.
-        If you don't need to handle actions, you can remove this method,
-        the hook created in __init__.py for it, the corresponding test,
-        and the actions.py file.
+        When dummy HTTP server Pebble is ready and the container is accessible, starts the
+        dummy HTTP server.
 
-        Learn more about actions at https://juju.is/docs/sdk/actions
+        Args:
+            event: Juju PebbleReadyEvent event
         """
-        fail = event.params["fail"]
-        if fail:
-            event.fail(fail)
+        if self._dummy_http_server_container.can_connect():
+            self._start_dummy_http_server()
         else:
-            event.set_results({"fortune": "A bug in the code is worth two in the documentation."})
+            self.unit.status = WaitingStatus(
+                f"Waiting for {self._dummy_http_server_container_name} container to be ready"
+            )
+            event.defer()
+
+    def _on_alertmanager_config_changed(self, _) -> None:
+        """Updates relation data bag with updated Alertmanager config."""
+        try:
+            alertmanager_config = RemoteConfigurationProvider.load_config_file(
+                self.ALERTMANAGER_CONFIG_FILE
+            )
+            self.remote_configuration_provider.update_relation_data_bag(alertmanager_config)
+        except ConfigReadError:
+            logger.warning("Error reading Alertmanager config file.")
```

Comment:
If the user provides a config that is not usable, we should probably do more than just log it.